crates/starknet-proxy-client/src/deploy.rs

```diff
@@ -0,0 +1,44 @@
+use std::sync::Arc;
+use ethers::abi::Tokenize;
+use ethers::contract::ContractError;
+use ethers::prelude::ContractInstance;
+use ethers::providers::ProviderError;
+use ethers::utils::hex::{self};
+use utils::LocalWalletSignerMiddleware;
+use ethereum_instance::deploy_contract;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    SerdeJson(#[from] serde_json::Error),
+    #[error("['bytecode']['object'] is not a string")]
+    BytecodeObject,
+    #[error(transparent)]
+    Hex(#[from] hex::FromHexError),
+    #[error("Failed to parse URL")]
+    UrlParser,
+    #[error(transparent)]
+    EthersContract(#[from] ContractError<LocalWalletSignerMiddleware>),
+    #[error(transparent)]
+    EthersProvider(#[from] ProviderError),
+    #[error("Invalid contract build artifacts: missing field `{0}`")]
+    ContractBuildArtifacts(&'static str),
+}
+
+const UNSAFE_PROXY: &str = include_str!("./artifacts/UnsafeProxy.json");
```

Comment:
I feel we should introduce a real proxy contract and keep this for testing (or throw it away as well).
Probably for another issue.

https://etherscan.io/address/0xb8fde0c30b1376293027835688e7926c03253142#code

Reply:
Good catch, this is the one used officially by Starknet - https://etherscan.io/address/0xc662c410C0ECf747543f5bA90660f6ABeBD9C8c4#code

@ocdbytes can you try adding this one and check if this doesn't break anything in e2e test cases.

Reply:
moving this to a new issue as it breaks test cases with Madara as well